akka-discovery/src/main/scala/akka/discovery/dns/DnsServiceDiscovery.scala

```diff
@@ -110,14 +120,17 @@ private[akka] class DnsServiceDiscovery(system: ExtendedActorSystem) extends Ser
     val mode = Srv
 
     def askResolve(): Future[Resolved] = {
-      dns.ask(DnsProtocol.Resolve(srvRequest, mode))(resolveTimeout).map {
-        case resolved: DnsProtocol.Resolved =>
-          log.debug("{} lookup result: {}", mode, resolved)
-          srvRecordsToResolved(srvRequest, resolved)
-        case resolved =>
-          log.warning("Resolved UNEXPECTED (resolving to Nil): {}", resolved.getClass)
-          Resolved(srvRequest, Nil)
-      }
+      dns
+        .ask(DnsProtocol.Resolve(srvRequest, mode))(resolveTimeout)
+        .map {
+          case resolved: DnsProtocol.Resolved =>
+            log.debug("{} lookup result: {}", mode, resolved)
+            srvRecordsToResolved(srvRequest, resolved)
+          case resolved =>
+            log.warning("Resolved UNEXPECTED (resolving to Nil): {}", resolved.getClass)
+            Resolved(srvRequest, Nil)
+        }
+        .transform(convertToTimeout)
```

Comment:
I guess we could `recoverWith`, though this is also :+1:

Reply:
:+1: will switch